providers/json-rpc-provider.js

```diff
@@ -65,7 +65,7 @@ utils.defineProperty(JsonRpcProvider.prototype, 'send', function(method, params)
     var request = {
         method: method,
         params: params,
-        id: 42,
+        id: this.chainId,
```

Comment:
This isn't the chain ID. It is the [JSON-RPC message id](http://www.jsonrpc.org/specification#request_object). Technically, we should change it with every request, but it is not necessary against in HTTP. In a WebSocket implementation we would (will) care about this.